core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -200,7 +201,31 @@ public MappedFieldType clone() {
         return new MappedFieldType(this);
     }
 
-    // norelease: we need to override freeze() and add safety checks that all settings are actually set
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof MappedFieldType)) return false;
```

Comment:
I generated all equals/hashcode methods with intellij. I've removed the checks before super.equals, and filed an issue with intellij [1] to be smarter about this.

I also had noticed always using Objects.equals here, even for primitive types. It looks like this is fixed [2] in an unreleased version of intellij. For now, I've gone through and changed each primitive type to use `==` instead.

[1] https://youtrack.jetbrains.com/issue/IDEA-141479
[2] https://youtrack.jetbrains.com/issue/IDEA-139622